src/container/srv_container.c

```diff
@@ -1426,8 +1426,11 @@ belongs_to_user(d_iov_t *key, struct find_hdls_by_cont_arg *arg)
 
 	rc = ds_sec_creds_are_same_user(&cred, arg->fha_cred);
 
-	if (pool_hdl == NULL)
+	if (pool_hdl)
+		ds_pool_hdl_put(pool_hdl);
```

Comment:
My bad. Sorry!

By the way, would you mind stop changing `p == NULL` or `p != NULL` into `!p` or `p` in similar cases, please? Thank you.

Reply:
ah, sure. yeah, I definitely agree p == NULL and p!= NULL is clearer.

Reply:
Thanks. Appreciated.